src/detect-pcre.h

```diff
@@ -39,6 +39,8 @@
 typedef struct DetectPcreData_ {
     /* pcre options */
     DetectParseRegex parse_regex;
+
+    int thread_ctx_id;
```

Comment:
Documentation missing. I would also suggest adding `jit` to the name to indicate it's only used by jit support. Should it also be in `PCRE_HAVE_JIT_EXEC` guards?